lms/djangoapps/teams/serializers.py

```diff
@@ -108,8 +112,43 @@ class Meta(object):
         read_only_fields = ("date_joined",)
 
 
-class TopicSerializer(serializers.Serializer):
-    """Serializes a topic."""
+class BaseTopicSerializer(serializers.Serializer):
+    """Serializes a topic without team_count."""
     description = serializers.CharField()
     name = serializers.CharField()
     id = serializers.CharField()  # pylint: disable=invalid-name
+
+
+class TopicSerializer(BaseTopicSerializer):
+    """
+    Adds team_count to the basic topic serializer.  Use only when
+    serializing a single topic.  When serializing many topics, use
+    `PaginatedTopicSerializer` to avoid O(N) SQL queries.
+    """
+    team_count = serializers.SerializerMethodField('get_team_count')
+
+    def get_team_count(self, topic):
+        """Get the number of teams associated with this topic"""
+        return CourseTeam.objects.filter(topic_id=topic['id']).count()
+
+
+class PaginatedTopicSerializer(PaginationSerializer):
```

Comment:
That makes sense. I'll add a class to test this serializer. There are two points you bring up which I'm not sure we have to test:
* For testing "what happens if there are counts for a topic that isn't being returned " - are you asking about the case in which topics with counts exist, but not for the current page? If so, we shouldn't have to test that; DRF's serializers only are aware of the page you give them.
* Again, testing DRF's ability to serialize a page ("verify that the pagination logic works") sounds to me like handing the serializer a page and expecting that it didn't serialize any data outside of the page. Why would it?